### PR TITLE
(PC-9637) Chunking .main.js = code splitting Web app

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-9637

Also include : 

- https://passculture.atlassian.net/browse/PC-9607 
- 
## Checklist

I have:

- [x] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [x] Made sure my feature is working on the relevant real / virtual devices.
- [x] Written **unit tests** for my feature.
- [x] Written **documentation on Notion** for my feature.
- [x] Added new reusable components to the AppComponents page and communicate to the team on slack
- [x] Added a **screenshot** for UI tickets.
- [x] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/libs/environment/env.web.ts [hunk 1]
```diff
@@ -6,5 +6,5 @@ import { Environment } from '.'
 
 // TODO: FIXME: configure cors on backend and disable proxy in webpackDevServer.config.js#L110
 export const env = (parseBooleanVariables(
-  omit(process.env, ['API_BASE_URL'])
+  omit(process.env, __DEV__ ? ['API_BASE_URL'] : [])
```
Comment (AnoukHello) [inline src/libs/environment/env.web.ts:9]: Pourquoi on fait ça ?

File: src/libs/environment/env.web.ts [hunk 2]
```diff
@@ -6,5 +6,5 @@ import { Environment } from '.'
 
 // TODO: FIXME: configure cors on backend and disable proxy in webpackDevServer.config.js#L110
```
Comment (AnoukHello) [inline src/libs/environment/env.web.ts:7]: Ce serait cool d'ajouter un nom ou un numéro de ticket à ce TODO

File: src/theme/index.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { theme as idCheckTheme } from '@pass-culture/id-check/src/theme'
+import deepmerge from 'deepmerge'
+import { DefaultTheme } from 'styled-components/native'
+import './styled.d'
+
+const theme: DefaultTheme = deepmerge(idCheckTheme, {})
+
+export { theme }
```
Comment (AnoukHello) [inline src/theme/index.ts:8]: Ce serait pas mieux de définir le theme dans l'app plutôt que de le récupérer de idcheck ? Genre si un jour on décide de ne plus utiliser idcheck

File: src/libs/environment/env.web.ts (hunk 1 above)
Comment (kopax-polyconseil) [reply src/libs/environment/env.web.ts:9]: Pour l'instant on a pas le bucket testing, donc on a pas pu configurer la communication CORS, cela nous permet d'utiliser fetch à travers un proxy node et localhost.

File: src/libs/environment/env.web.ts (hunk 2 above)
Comment (kopax-polyconseil) [reply src/libs/environment/env.web.ts:7]: OK

File: src/theme/index.ts (hunk 3 above)
Comment (kopax-polyconseil) [reply src/theme/index.ts:8]: > C'est le cas,

notre theme étend de id check, d'ou le deepmerge, c'est juste que on a 0 variable coté app native, d'ou l'objet vide.

File: src/libs/environment/env.web.ts (hunk 2 above)
Comment (kopax-polyconseil) [reply src/libs/environment/env.web.ts:7]: https://passculture.atlassian.net/browse/PC-9847

**Review by Lucasbeneston — APPROVED** [review]

**Pull request merged**